packages/next/pages/_document.tsx

```diff
@@ -970,24 +970,30 @@ export class NextScript extends Component<OriginProps> {
     const { __NEXT_DATA__, largePageDataBytes } = context
     try {
       const data = JSON.stringify(__NEXT_DATA__)
-      const bytes =
-        process.env.NEXT_RUNTIME === 'edge'
-          ? new TextEncoder().encode(data).buffer.byteLength
-          : Buffer.from(data).byteLength
-      const prettyBytes = require('../lib/pretty-bytes').default
 
-      if (largePageDataBytes && bytes > largePageDataBytes) {
-        console.warn(
-          `Warning: data for page "${__NEXT_DATA__.page}"${
-            __NEXT_DATA__.page === context.dangerousAsPath
-              ? ''
-              : ` (path "${context.dangerousAsPath}")`
-          } is ${prettyBytes(
-            bytes
-          )} which exceeds the threshold of ${prettyBytes(
-            largePageDataBytes
-          )}, this amount of data can reduce performance.\nSee more info here: https://nextjs.org/docs/messages/large-page-data`
-        )
+      if (
+        process.env.NODE_ENV === 'development' ||
```

Comment:
I don't think we want to revert to only warn during development, as this was intentionally switched in #37264 to warn at build-time as well, based on feedback.

Reply:
@balazsorban44 
Yes, that's why I use `||` to make it work at build time.
Since __NEXT_DATA__.gssp and __NEXT_DATA__.gip values ​​do not exist at build time, the corresponding branch is always executed at build time.
I chose the next best thing because there are no build flags.